src/components/backup/BackupIcloudStep.js

```diff
@@ -0,0 +1,387 @@
+  const dispatch = useDispatch();
+  const [validPassword, setValidPassword] = useState(false);
+  const [passwordFocused, setPasswordFocused] = useState(true);
+  const [password, setPassword] = useState(loadedPassword);
+  const [confirmPassword, setConfirmPassword] = useState(loadedPassword);
+
+  const [label, setLabel] = useState(
+    !validPassword ? '􀙶 Add to iCloud Backup' : '􀎽 Confirm Backup'
+  );
+  const passwordRef = useRef();
+  const confirmPasswordRef = useRef();
+
+  useEffect(() => {
+    setTimeout(() => {
+      passwordRef.current?.focus();
+    }, 1);
+  }, []);
+
+  const onPasswordFocus = useCallback(() => {
+    setPasswordFocused(true);
+    currentlyFocusedInput.current = passwordRef.current;
+  }, []);
+
+  const onConfirmPasswordFocus = useCallback(() => {
+    currentlyFocusedInput.current = confirmPasswordRef.current;
+  }, []);
+
+  const onPasswordBlur = useCallback(() => {
+    setPasswordFocused(false);
+  }, []);
+
+  const onPasswordSubmit = useCallback(() => {
+    confirmPasswordRef.current.focus();
+  }, []);
+
+  useEffect(() => {
+    let passwordIsValid = false;
+    if (
+      password === confirmPassword &&
+      password !== '' &&
+      password.length >= 8
+    ) {
+      passwordIsValid = true;
+    }
+
+    let newLabel = '';
+    if (passwordIsValid) {
+      newLabel = loadedPassword ? '􀙶 Add to iCloud Backup' : '􀎽 Confirm Backup';
+    } else if (password.length < 8) {
+      newLabel = 'Minimum 8 characters';
+    } else if (
+      password !== '' &&
+      password.length < 8 &&
+      !passwordRef.current.isFocused()
+    ) {
+      newLabel = 'Use a longer password';
+    } else if (
+      password.length >= 8 &&
+      confirmPassword.length >= 8 &&
+      confirmPassword.length >= password.length &&
+      password !== confirmPassword
+    ) {
+      newLabel = `Passwords don't match`;
+    } else if (password.length >= 8 && !passwordFocused) {
+      newLabel = 'Confirm password';
+    } else if (password.length >= 8 && passwordFocused) {
+      const passInfo = zxcvbn(password);
+      switch (passInfo.score) {
+        case 0:
+        case 1:
+          newLabel = '💩 Weak password';
+          break;
+        case 2:
+          newLabel = '💪 Good password';
+          break;
+        case 3:
+          newLabel = '🦾 Great password';
+          break;
+        case 4:
+          newLabel = '🏰️ Strong password';
+          break;
+        default:
+      }
+    }
+
+    setValidPassword(passwordIsValid);
+    setLabel(newLabel);
+  }, [confirmPassword, loadedPassword, password, passwordFocused]);
+
+  const onPasswordChange = useCallback(
+    ({ nativeEvent: { text: inputText } }) => {
+      setPassword(inputText);
+    },
+    []
+  );
+
+  const onConfirmPasswordChange = useCallback(
+    ({ nativeEvent: { text: inputText } }) => {
+      setConfirmPassword(inputText);
+    },
+    []
+  );
+  const onConfirmBackup = useCallback(async () => {
```

Comment:
`onConfirmBackup` is virtually the same as BackupConfirmPasswordStep's `onSubmit` function - can we pull this out into its own hook? Differences include which input ref needs to be refocused on a failure, which can be passed in as a function handling that